cmd/fts/cmd/run.go

```diff
@@ -0,0 +1,170 @@
+package cmd
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/gosuri/uiprogress"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	dryRun         bool
+	mainnet        bool
+	maxStagedBytes int64
+	maxDealBytes   int64
+	minDealBytes   int64
+	hiddenFiles    bool
+	taskFolder     string
+	resultsOut     string
+	concurrent     int64
+	ipfsrevproxy   string
+	retryErrors    bool
+)
+
+func init() {
+	runCmd.Flags().BoolVarP(&dryRun, "dry-run", "d", false, "Run through steps without pushing data to Powergate API.")
+	runCmd.Flags().BoolVarP(&mainnet, "mainnet", "n", false, "Sets staging limits based on localnet or mainnet.")
+	runCmd.Flags().BoolVarP(&retryErrors, "retry", "e", false, "Retry tasks with error status")
+	runCmd.Flags().StringVar(&taskFolder, "folder", "", "Folder with organized tasks of directories or files")
+	runCmd.Flags().BoolVarP(&hiddenFiles, "all", "a", false, "Include hidden files & folders from top level folder")
+	runCmd.Flags().StringVar(&ipfsrevproxy, "ipfsrevproxy", "127.0.0.1:6002", "Powergate IPFS reverse proxy multiaddr")
+	runCmd.Flags().StringVarP(&resultsOut, "results", "r", "results.json", "The location to store intermediate and final results.")
+
+	// Not included in the public commands
+	runCmd.Flags().Int64Var(&concurrent, "concurrent", 10000, "Max concurrent tasks being processed")
+	runCmd.Flags().Int64Var(&maxStagedBytes, "max-staged-bytes", 0, "Maximum bytes of all tasks queued on staging")
+	runCmd.Flags().Int64Var(&maxDealBytes, "max-deal-bytes", 0, "Maximum bytes of a single deal")
+	runCmd.Flags().Int64Var(&minDealBytes, "min-deal-bytes", 0, "Minimum bytes of a single deal")
+	runCmd.Flags().MarkHidden("concurrent")
+	runCmd.Flags().MarkHidden("max-staged-bytes")
+	runCmd.Flags().MarkHidden("max-deal-bytes")
+	runCmd.Flags().MarkHidden("min-deal-bytes")
+
+	rootCmd.AddCommand(runCmd)
+}
+
+var runCmd = &cobra.Command{
+	Use:   "run",
+	Short: "Run a storage task pipeline",
+	Long:  `Run a storage task pipeline to migrate large collections of data to Filecoin`,
+	PreRun: func(cmd *cobra.Command, args []string) {
+		err := viper.BindPFlags(cmd.Flags())
+		checkErr(err)
+		// set default variables
+		if mainnet {
```

Comment:
couldn't find a good way in cobra to manage two sets of options like this, so just did the logic